File: keevo/backend/src/main/java/com/keevo/identity/auth/domain/port/in/RegistrationResult.java
package com.keevo.identity.auth.domain.port.in;

/**
 * RegistrationResult — Output of a successful user registration.
 *
 * <p>Pure Java record — NO HTTP/Spring/framework types.
 */
public record RegistrationResult(
    String tenantCode,
    String token,
    String userId,
    String tenantId
) {}

File: keevo/app/lib/features/auth/domain/model/registration_result.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'registration_result.freezed.dart';

/// RegistrationResult — Domain model for a successful registration response.
///
/// Immutable Freezed value object — auto-generates equality, hashCode, copyWith and toString.
/// Mirrors the backend RegistrationResult record.
@freezed
class RegistrationResult with _$RegistrationResult {
  const factory RegistrationResult({
    required String tenantCode,
    required String token,
    required String userId,
    required String tenantId,
  }) = _RegistrationResult;
}
